parser/src/parser.rs

```diff
@@ -630,18 +629,15 @@ where
     /// [1]: https://en.wikipedia.org/wiki/Shunting-yard_algorithm
     /// [2]: https://en.wikipedia.org/wiki/Reverse_Polish_notation#Postfix_evaluation_algorithm
     /// [3]: https://eli.thegreenplace.net/2012/08/02/parsing-expressions-by-precedence-climbing
-    fn parse_binary_expression(&mut self) -> BoxedResult<'f, ast::Expr<'f>> {
+    fn parse_binary_expression(&mut self) -> ParserResult<'f, ast::Expr<'f>> {
         let mut operator_stack = Vec::new();
         let mut operand_stack = Vec::new();
-        fn rpn_eval<'f>(
-            operand_stack: &mut Vec<Box<Spanned<'f, ast::Expr<'f>>>>,
```

Comment:
Clippy wanted me to remove this `Box`. So I did. This led to a huge fallout. But by the time I realized how big the fallout was, I was in too deep already.